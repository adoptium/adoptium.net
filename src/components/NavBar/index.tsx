'use client'

import React, { useState, useEffect, Fragment } from "react"
import { Link, usePathname } from "@/i18n/navigation"
import Image from "next/image"
import {
  Dialog,
  DialogPanel,
  Menu,
  MenuButton,
  MenuItem,
  MenuItems,
  Transition,
} from "@headlessui/react"
import { FaChevronDown, FaRegBell } from "react-icons/fa"
import { BsXLg, BsList } from "react-icons/bs"

import IconSocial from "@/components/IconSocial"
import LanguageSelector from "@/components/LanguageSelector"
import Announcements from "@/components/Announcements"

interface NavItem {
  name: string
  href?: string
  children?: NavItem[]
}

function classNames(...classes: string[]) {
  return classes.filter(Boolean).join(" ")
}

const navigation: NavItem[] = [
  { name: "Join Us", href: "/support-us" },
  { name: "Latest Releases", href: "/temurin/releases" },
  { name: "Marketplace", href: "/marketplace" },
  {
    name: "Projects",
    children: [
      { name: "Eclipse Temurin", href: "/temurin" },
      { name: "Eclipse AQAvit", href: "/aqavit" },
      { name: "Eclipse Mission Control", href: "/jmc" },
      { name: "Eclipse Migration Toolkit", href: "/emt" },
    ],
  },
  {
    name: "Resources",
    children: [
      { name: "Status", href: "https://status.adoptium.net" },
      { name: "Release Notes", href: "/temurin/release-notes" },
      { name: "Installation Guide", href: "/installation" },
      { name: "Documentation", href: "/docs" },
      { name: "FAQs", href: "/docs/faq" },
      { name: "Brand & Promotion", href: "/docs/logo-styleguide" },
    ],
  },
  {
    name: "Community",
    children: [
      { name: "Support", href: "/support" },
      { name: "News", href: "/news" },
      { name: "Events", href: "/events" },
      { name: "Slack", href: "/slack" },
    ],
  },
]

/**
 * A reusable mobile link component that renders a Link (if the href starts with "/")
 * or a regular <a> element. Both get the same classes.
 */
const MobileLink: React.FC<{
  href?: string
  name: string
  activePaths: Set<string>
  onClick?: () => void
}> = ({ href, name, activePaths, onClick }) => {
  const commonClasses = classNames(
    "-mx-3 block rounded-lg px-3 py-2 text-[20px] font-normal leading-7 text-white-900 hover:bg-white-50",
    href && activePaths.has(href) ? "text-rose-600" : "",
  )

  if (href && href.startsWith("/")) {
    return (
      <Link href={href} className={commonClasses} onClick={onClick}>
        {name}
      </Link>
    )
  }
  return (
    <a href={href} className={commonClasses} onClick={onClick}>
      {name}
    </a>
  )
}

/**
 * A tiny divider used between mobile nav items.
 */
const MobileDivider: React.FC = () => (
  <div className="w-full px-3 bg-[#3E3355] h-[1px]"></div>
)

function calculateActivePaths(currentPath: string): Set<string> {
  if (!currentPath) return new Set();

  // Remove locale prefix from current path (e.g., /en-GB/docs/faq -> /docs/faq)
  const removeLocalePrefix = (pathname: string): string => {
    // Match patterns like /en-GB/, /de/, /es/, /fr/, /zh-CN/, etc.
    const localePattern = /^\/[a-z]{2}(-[A-Z]{2})?\//
    return pathname.replace(localePattern, '/')
  }

  const pathWithoutLocale = removeLocalePrefix(currentPath);
  const normalizedCurrentPath = pathWithoutLocale.endsWith('/') ? pathWithoutLocale.slice(0, -1) : pathWithoutLocale;

  // Get all navigation paths to find the most specific match
  const allPaths: { path: string, normalizedPath: string }[] = [];

  navigation.forEach(item => {
    if (item.href) {
      const normalizedPath = item.href.endsWith('/') ? item.href.slice(0, -1) : item.href;
      allPaths.push({ path: item.href, normalizedPath });
    }
    if (item.children) {
      item.children.forEach(child => {
        if (child.href) {
          const normalizedPath = child.href.endsWith('/') ? child.href.slice(0, -1) : child.href;
          allPaths.push({ path: child.href, normalizedPath });
        }
      })
    }
  });

  // Find all matching paths
  const matchingPaths = allPaths.filter(({ normalizedPath }) =>
    normalizedCurrentPath === normalizedPath || normalizedCurrentPath.startsWith(normalizedPath + '/')
  );

  // If no matches, return empty set
  if (matchingPaths.length === 0) return new Set();

  // Find the longest (most specific) matching path
  const longestMatch = matchingPaths.reduce((longest, current) =>
    current.normalizedPath.length > longest.normalizedPath.length ? current : longest
  );

  // Only highlight the most specific match
  return new Set([longestMatch.path]);
}

const NavBar = ({ locale }: { locale: string }) => {
  const [mobileMenuOpen, setMobileMenuOpen] = useState(false)
  const [showLastSlide, setShowLastSlide] = useState(false)
  const [showAnnouncement, setShowAnnouncement] = useState(false)
  const [activeLastSlide, setActiveLastSlide] = useState<NavItem | null>(null)
  // To prevent hydration mismatch, we'll start with no active paths
  const [activePaths, setActivePaths] = useState<Set<string>>(new Set())
  const pathname = usePathname();

  useEffect(() => {
    if (!mobileMenuOpen) {
      setShowLastSlide(false)
    }
  }, [mobileMenuOpen])

  // Calculate active paths on client-side only after initial render
  useEffect(() => {
    if (typeof window !== "undefined") {
      const paths = calculateActivePaths(pathname);
      setActivePaths(paths);
    }
  }, [pathname]);

  const openLastSlideHandler = (item: NavItem) => {
    setShowLastSlide(true)
    setActiveLastSlide(item)
  }

  const [scrolled, setScrolled] = useState(false)

  useEffect(() => {
    const handleScroll = () => {
      setScrolled(window.scrollY > 0)
    }

    window.addEventListener("scroll", handleScroll)
    return () => {
      window.removeEventListener("scroll", handleScroll)
    }
  }, [])

  return (
    <header
      className={`sticky inset-x-0 top-0 z-50 ${scrolled
        ? "bg-[#200E46] border-b-2 border-[#3E3355]/85 backdrop-blur-xl"
        : ""
        }`}
    >
      {showAnnouncement && (
        <Announcements handleClose={() => setShowAnnouncement(false)} />
      )}
      {/* Container div to center the nav content */}
      <div className="max-w-[1288px] w-full mx-auto px-3">
        <nav
          className="flex items-center gap-5 justify-between py-6"
          aria-label="Global"
        >
          <Link href="/">
            <Image
              src="/images/adoptium-logo-dark.svg"
              alt="Adoptium Logo"
              width={0}
              height={0}
              className="h-10 w-auto"
            />
          </Link>
          <div className="flex items-center gap-4">
            <div className="hidden lg:flex lg:gap-6 xl:gap-x-12">
              {navigation.map(item =>
                item.children ? (
                  <Menu
                    as="div"
                    key={`desktop-${item.name}`}
                    className="relative inline-block text-left"
                  >
                    <div>
                      <MenuButton className="inline-flex w-full gap-2 justify-center rounded-md text-sm font-semibold text-white-900 hover:bg-white-50">
                        {item.name}
                        <FaChevronDown
                          className="-mr-1 mt-1"
                          aria-hidden="true"
                        />
                      </MenuButton>
                    </div>
                    <Transition
                      as={Fragment}
                      enter="transition ease-out duration-100"
                      enterFrom="transform opacity-0 scale-95"
                      enterTo="transform opacity-100 scale-100"
                      leave="transition ease-in duration-75"
                      leaveFrom="transform opacity-100 scale-100"
                      leaveTo="transform opacity-0 scale-95"
                    >
                      <MenuItems
                        className="absolute right-0 z-10 mt-2 origin-top-right rounded-md bg-[#0E002A] shadow-lg ring-1 ring-black/5 focus:outline-hidden"
                        style={{ minWidth: "max-content" }}
                      >
                        <div className="py-6 px-4">
                          {item.children.map(child => (
                            <MenuItem key={`mobile-${child.name}`} as="div">
                              {() => (
                                <MobileLink
                                  href={child.href}
                                  name={child.name}
                                  activePaths={activePaths}
                                />
                              )}
                            </MenuItem>
                          ))}
                        </div>
                      </MenuItems>
                    </Transition>
                  </Menu>
                ) : item.href ? (
                  <Link
                    key={`desktop-${item.name}`}
                    href={item.href}
                    className={classNames(
                      "text-sm font-semibold leading-6 text-white-900",
                      item.href && activePaths.has(item.href) ? "text-rose-600" : ""
                    )}
                  >
                    {item.name}
                  </Link>
                ) : (
                  <a
                    key={`desktop-${item.name}`}
                    href={item.href}
                    className={classNames(
                      "text-sm font-semibold leading-6 text-white-900",
                      item.href && activePaths.has(item.href) ? "text-rose-600" : ""
                    )}
                  >
                    {item.name}
                  </a>
                ),
              )}
            </div>
            <div className="flex space-x-3 h-12">
              <div className="hidden sm:block">
                <LanguageSelector locale={locale} />
              </div>
              <div className="p-3 h-full rounded-3xl border-2 border-gray-700 justify-start items-center gap-3 inline-flex cursor-pointer">
                <div
                  aria-label="notifications"
                  onClick={() => setShowAnnouncement(!showAnnouncement)}
                  className="relative"
                >
                  <FaRegBell size={20} />
                  <span className="absolute -top-1 -right-1 h-4 w-4 bg-red-500 rounded-full text-xs text-white font-bold flex items-center justify-center">
                    1
                  </span>
                </div>
              </div>
            </div>
            <div className="flex lg:hidden ml-3">
              <button
                type="button"
                className="-m-2.5 inline-flex items-center justify-center rounded-md text-white-700"
                onClick={() => setMobileMenuOpen(true)}
              >
                <span className="sr-only">Open main menu</span>
                <div className="border-2 border-[#3e3355] p-3 rounded-full">
                  <BsList className="text-2xl" />
                </div>
              </button>
            </div>
          </div>
        </nav>
      </div>
      <Dialog
        as="div"
        className="lg:hidden"
        open={mobileMenuOpen}
        onClose={setMobileMenuOpen}
      >
        <div className="fixed inset-0 z-50" />
        <DialogPanel className="fixed flex-col h-full flex inset-y-0 right-0 z-50 w-full overflow-y-auto bg-purple px-6 py-6 sm:max-w-sm sm:ring-1 sm:ring-white-900/10">
          <div className="flex items-center justify-between">
            <Link href="/" className="-m-1.5 p-1.5">
              <span className="sr-only">Eclipse Adoptium</span>
              <Image
                className="h-8 mb-0 w-auto"
                src="/images/adoptium-logo-dark.svg"
                alt="Adoptium Logo"
                width={0}
                height={0}
              />
            </Link>
            <button
              type="button"
              className="-m-2.5 rounded-md p-2.5 text-white-700"
              onClick={() => setMobileMenuOpen(false)}
            >
              <span className="sr-only">Close menu</span>
              <div className="border-2 border-[#3e3355] p-3 rounded-full">
                <BsXLg className="text-xl" />
              </div>
            </button>
          </div>
          <div className="mt-6 grow relative w-full h-full overflow-hidden flow-root">
            {/* Mobile menu – main navigation */}
            <div
              className={`-my-6 absolute duration-200 h-full left-0 w-full divide-y divide-white-500/10 ${showLastSlide ? "left-[-100%]" : ""
                }`}
            >
              <div className="space-y-2 py-6">
                {navigation.map((item, index) => (
                  <div key={item.name}>
                    {item.children ? (
                      <div
                        onClick={() => openLastSlideHandler(item)}
                        className="flex items-center justify-between"
                      >
                        <div className="-mx-3 block rounded-lg px-3 py-2 text-[20px] font-normal leading-7 text-white-900 hover:bg-white-50">
                          {item.name}
                        </div>
                        <svg
                          width="8"
                          height="12"
                          viewBox="0 0 8 12"
                          fill="none"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M1.5 11L6.5 6L1.5 1"
                            stroke="white"
                            strokeWidth="1.5"
                            strokeLinecap="round"
                            strokeLinejoin="round"
                          />
                        </svg>
                      </div>
                    ) : (
                      <div className="flex justify-between">
                        <MobileLink
                          href={item.href}
                          name={item.name}
                          activePaths={activePaths}
                          onClick={() => setMobileMenuOpen(false)}
                        />
                      </div>
                    )}
                    {navigation.length !== index + 1 && <MobileDivider />}
                  </div>
                ))}
              </div>
            </div>

            {/* Mobile menu – last slide (child links) */}
            <div
              className={`absolute duration-200 w-full h-full ${showLastSlide ? "left-0" : "left-full"
                }`}
            >
              <div
                onClick={() => setShowLastSlide(false)}
                className="flex items-center space-x-3"
              >
                <svg
                  width="48"
                  height="48"
                  viewBox="0 0 48 48"
                  fill="none"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <rect
                    x="1"
                    y="1"
                    width="46"
                    height="46"
                    rx="23"
                    stroke="#3E3355"
                    strokeWidth="2"
                  />
                  <path
                    d="M31 24H17"
                    stroke="white"
                    strokeWidth="1.5"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                  />
                  <path
                    d="M24 31L17 24L24 17"
                    stroke="white"
                    strokeWidth="1.5"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                  />
                </svg>
                <span className="text-[20px] text-white font-[400]">
                  {activeLastSlide && activeLastSlide.name}
                </span>
              </div>

              <div className="space-y-2 w-full py-6">
                {activeLastSlide &&
                  activeLastSlide.children?.map((item, index) => (
                    <div key={index}>
                      <div className="flex w-full justify-between">
                        <MobileLink
                          href={item.href}
                          name={item.name}
                          activePaths={activePaths}
                          onClick={() => setMobileMenuOpen(false)}
                        />
                      </div>
                      {activeLastSlide.children &&
                        index !== activeLastSlide.children.length - 1 && (
                          <MobileDivider />
                        )}
                    </div>
                  ))}
              </div>
            </div>
          </div>

          {/* Social media and announcement */}
          {!showLastSlide && (
            <div className="border-t-[1px] border-[#3E3355] space-y-3 pt-6">
              <div className="flex items-center justify-between">
                <IconSocial
                  link="https://twitter.com/adoptium"
                  className="text-2xl"
                  icon="twitter"
                />
                <IconSocial
                  link="https://www.linkedin.com/showcase/adoptium/"
                  className="text-2xl"
                  icon="linkedin"
                />
                <IconSocial
                  link="https://www.youtube.com/c/EclipseAdoptium"
                  className="text-2xl"
                  icon="youtube"
                />
                <IconSocial
                  link="https://github.com/adoptium"
                  className="text-2xl"
                  icon="github"
                />
                <IconSocial
                  link="https://adoptium.slack.com/"
                  className="text-2xl"
                  icon="slack"
                />
                <IconSocial
                  link="https://bsky.app/profile/adoptium.net"
                  className="text-2xl"
                  icon="bluesky"
                />
                <IconSocial
                  link="https://fosstodon.org/@adoptium"
                  className="text-2xl"
                  icon="mastodon"
                />
              </div>
              <div className="sm:hidden w-full mt-6">
                <LanguageSelector locale={locale} />
              </div>
            </div>
          )}
        </DialogPanel>
      </Dialog>
    </header>
  )
}

export default NavBar
