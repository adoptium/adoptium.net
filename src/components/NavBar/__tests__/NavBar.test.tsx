// Some portions generated by Co-Pilot
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { describe, it, expect, afterEach, vi } from 'vitest';
import NavBar from '../index';

vi.mock("next-intl/navigation", () => ({
  createNavigation: () => ({
    Link: () => null,
    redirect: () => {},
    usePathname: () => "",
    useRouter: () => ({}),
  }),
}));

describe('NavBar', () => {
    afterEach(() => {
        cleanup();
    });

    it('should match the snapshot', () => {
        const { container } = render(<NavBar locale='en' />);
        expect(container.firstChild).toMatchSnapshot();
    });
});
