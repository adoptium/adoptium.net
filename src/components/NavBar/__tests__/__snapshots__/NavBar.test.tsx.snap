// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar > should match the snapshot 1`] = `
<header
  class="sticky inset-x-0 top-0 z-50 "
>
  <div
    class="max-w-[1288px] w-full mx-auto px-3"
  >
    <nav
      aria-label="Global"
      class="flex items-center gap-5 justify-between py-6"
    >
      <a
        href="/"
      >
        <img
          alt="Adoptium Logo"
          class="h-10 w-auto"
          data-nimg="1"
          decoding="async"
          height="0"
          loading="lazy"
          src="/images/adoptium-logo-dark.svg"
          style="color: transparent;"
          width="0"
        />
      </a>
      <div
        class="flex items-center gap-4"
      >
        <div
          class="hidden lg:flex lg:gap-6 xl:gap-x-12"
        >
          <div
            class="relative inline-block text-left"
            data-headlessui-state=""
          >
            <div>
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="inline-flex w-full gap-2 justify-center rounded-md text-sm font-semibold text-white-900 hover:bg-white-50 cursor-pointer"
                data-headlessui-state=""
                id="headlessui-menu-button-«r2»"
                type="button"
              >
                About Us
                <svg
                  aria-hidden="true"
                  class="-mr-1 mt-1"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <a
            class="text-sm font-semibold leading-6 text-white-900"
            href="/temurin/releases"
          >
            Latest Releases
          </a>
          <a
            class="text-sm font-semibold leading-6 text-white-900"
            href="/marketplace"
          >
            Marketplace
          </a>
          <div
            class="relative inline-block text-left"
            data-headlessui-state=""
          >
            <div>
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="inline-flex w-full gap-2 justify-center rounded-md text-sm font-semibold text-white-900 hover:bg-white-50 cursor-pointer"
                data-headlessui-state=""
                id="headlessui-menu-button-«r5»"
                type="button"
              >
                Projects
                <svg
                  aria-hidden="true"
                  class="-mr-1 mt-1"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="relative inline-block text-left"
            data-headlessui-state=""
          >
            <div>
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="inline-flex w-full gap-2 justify-center rounded-md text-sm font-semibold text-white-900 hover:bg-white-50 cursor-pointer"
                data-headlessui-state=""
                id="headlessui-menu-button-«r8»"
                type="button"
              >
                Resources
                <svg
                  aria-hidden="true"
                  class="-mr-1 mt-1"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div
            class="relative inline-block text-left"
            data-headlessui-state=""
          >
            <div>
              <button
                aria-expanded="false"
                aria-haspopup="menu"
                class="inline-flex w-full gap-2 justify-center rounded-md text-sm font-semibold text-white-900 hover:bg-white-50 cursor-pointer"
                data-headlessui-state=""
                id="headlessui-menu-button-«rb»"
                type="button"
              >
                Community
                <svg
                  aria-hidden="true"
                  class="-mr-1 mt-1"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="flex space-x-3 h-12"
        >
          <div
            class="hidden sm:block"
          >
            <div
              class="relative inline-block text-left"
              data-headlessui-state=""
            >
              <div>
                <button
                  aria-expanded="false"
                  aria-haspopup="menu"
                  class="inline-flex w-full justify-center gap-3 rounded-3xl bg-transparent h-full px-4 py-3 text-sm text-white shadow-xs border-2 border-gray-700"
                  data-headlessui-state=""
                  id="headlessui-menu-button-«re»"
                  type="button"
                >
                  <img
                    class="mb-0 h-5"
                    src="data:image/svg+xml,%3csvg xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' viewBox='0 0 7410 3900'%3e%3cpath fill='%23b22234' d='M0 0h7410v3900H0z'/%3e%3cpath stroke='%23fff' stroke-width='300' d='M0 450h7410m0 600H0m0 600h7410m0 600H0m0 600h7410m0 600H0'/%3e%3cpath fill='%233c3b6e' d='M0 0h2964v2100H0z'/%3e%3cg fill='%23fff'%3e%3cg id='d'%3e%3cg id='c'%3e%3cg id='e'%3e%3cg id='b'%3e%3cpath id='a' d='m247 90 70.534 217.082-184.66-134.164h228.253L176.466 307.082z'/%3e%3cuse xlink:href='%23a' y='420'/%3e%3cuse xlink:href='%23a' y='840'/%3e%3cuse xlink:href='%23a' y='1260'/%3e%3c/g%3e%3cuse xlink:href='%23a' y='1680'/%3e%3c/g%3e%3cuse xlink:href='%23b' x='247' y='210'/%3e%3c/g%3e%3cuse xlink:href='%23c' x='494'/%3e%3c/g%3e%3cuse xlink:href='%23d' x='988'/%3e%3cuse xlink:href='%23c' x='1976'/%3e%3cuse xlink:href='%23e' x='2470'/%3e%3c/g%3e%3c/svg%3e"
                  />
                  <span>
                    English
                  </span>
                  <svg
                    aria-hidden="true"
                    class="-mr-1 h-5 w-5 text-white"
                    fill="currentColor"
                    height="15"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 448 512"
                    width="15"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                    />
                  </svg>
                </button>
              </div>
            </div>
          </div>
          <div
            class="p-3 h-full rounded-3xl border-2 border-gray-700 justify-start items-center gap-3 inline-flex cursor-pointer"
          >
            <div
              aria-label="notifications"
              class="relative"
            >
              <svg
                fill="currentColor"
                height="20"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M439.39 362.29c-19.32-20.76-55.47-51.99-55.47-154.29 0-77.7-54.48-139.9-127.94-155.16V32c0-17.67-14.32-32-31.98-32s-31.98 14.33-31.98 32v20.84C118.56 68.1 64.08 130.3 64.08 208c0 102.3-36.15 133.53-55.47 154.29-6 6.45-8.66 14.16-8.61 21.71.11 16.4 12.98 32 32.1 32h383.8c19.12 0 32-15.6 32.1-32 .05-7.55-2.61-15.27-8.61-21.71zM67.53 368c21.22-27.97 44.42-74.33 44.53-159.42 0-.2-.06-.38-.06-.58 0-61.86 50.14-112 112-112s112 50.14 112 112c0 .2-.06.38-.06.58.11 85.1 23.31 131.46 44.53 159.42H67.53zM224 512c35.32 0 63.97-28.65 63.97-64H160.03c0 35.35 28.65 64 63.97 64z"
                />
              </svg>
              <span
                class="absolute -top-1 -right-1 h-4 w-4 bg-red-500 rounded-full text-xs text-white font-bold flex items-center justify-center"
              >
                1
              </span>
            </div>
          </div>
        </div>
        <div
          class="flex lg:hidden ml-3"
        >
          <button
            class="-m-2.5 inline-flex items-center justify-center rounded-md text-white-700"
            type="button"
          >
            <span
              class="sr-only"
            >
              Open main menu
            </span>
            <div
              class="border-2 border-[#3e3355] p-3 rounded-full"
            >
              <svg
                class="text-2xl"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.5 12a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5m0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5m0-4a.5.5 0 0 1 .5-.5h10a.5.5 0 0 1 0 1H3a.5.5 0 0 1-.5-.5"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </button>
        </div>
      </div>
    </nav>
  </div>
</header>
`;
